error by trying to get test cases of GCJ problems whose test cases are not combined to a single sample case

GCJでテストケースが複数存在する問題のダウンロードを行おうとするとエラーが発生します
OS:Windows10
online-judge-tools: 10.0.8
```
$ oj dl https://codingcompetitions.withgoogle.com/kickstart/round/000000000019ffc7/00000000001d3ff3
[ERROR] the number of <pre class="io-content"> is not two
```

--- a/onlinejudge/service/google.py
+++ b/onlinejudge/service/google.py
@@ -9,6 +9,7 @@ import json
 import re
 import string
 import urllib.parse
+from itertools import islice
 from typing import *
 
 import bs4
@@ -92,20 +93,28 @@ class GoogleCodeJamProblem(onlinejudge.type.Problem):
         # parse HTML
         soup = bs4.BeautifulSoup(statement, utils.html_parser)
         io_contents = soup.find_all('pre', class_='io-content')
-        if len(io_contents) != 2:
-            raise SampleParseError("""the number of <pre class="io-content"> is not two""")
-        if io_contents[0].text.startswith('Case #'):
-            log.warning('''the sample input starts with "Case #"''')
-        if not io_contents[1].text.startswith('Case #'):
-            log.warning('''the sample output doesn't start with "Case #"''')
-        sample = TestCase(
-            'sample',
-            'Input',
-            utils.textfile(io_contents[0].text.rstrip()).encode(),
-            'Output',
-            utils.textfile(io_contents[1].text.rstrip()).encode(),
-        )
-        return [sample]
+        if len(io_contents) % 2 != 0:
+            raise SampleParseError("""the number of <pre class="io-content"> is not multiple of two""")
+
+        input_contents = islice(io_contents, 0, None, 2)
+        output_contents = islice(io_contents, 1, None, 2)
+
+        samples = []
+
+        for index, (input_content, output_content) in enumerate(zip(input_contents, output_contents)):
+            if input_content.text.startswith('Case #'):
+                log.warning('''the sample input starts with "Case #"''')
+            if not output_content.text.startswith('Case #'):
+                log.warning('''the sample output doesn't start with "Case #"''')
+            samples.append(TestCase(
+                'sample-{}'.format(index + 1),
+                'Input {}'.format(index + 1),
+                utils.textfile(input_content.text.rstrip()).encode(),
+                'Output {}'.format(index + 1),
+                utils.textfile(output_content.text.rstrip()).encode(),
+            ))
+
+        return samples
 
     def get_url(self) -> str:
         if self.domain == 'codingcompetitions.withgoogle.com':

--- a/tests/service_google.py
+++ b/tests/service_google.py
@@ -68,10 +68,10 @@ class GoogleCodeJamProblemTest(unittest.TestCase):
             """).encode()
         self.assertEqual(problem.download_sample_cases(), [
             TestCase(
-                'sample',
-                'Input',
+                'sample-1',
+                'Input 1',
                 sample_input,
-                'Output',
+                'Output 1',
                 sample_output,
             ),
         ])
@@ -94,11 +94,62 @@ class GoogleCodeJamProblemTest(unittest.TestCase):
             """).encode()
         self.assertEqual(problem.download_sample_cases(), [
             TestCase(
-                'sample',
-                'Input',
+                'sample-1',
+                'Input 1',
                 sample_input,
-                'Output',
+                'Output 1',
                 sample_output,
+            ),
+        ])
+
+    def test_download_multiple_samples(self):
+        problem = GoogleCodeJamProblem.from_url('https://codingcompetitions.withgoogle.com/kickstart/round/000000000019ffc7/00000000001d3ff3')
+        sample_input1 = textwrap.dedent("""\
+            2
+            2 2
+            KICK
+            START
+            8 2
+            G
+            G
+            GO
+            GO
+            GOO
+            GOO
+            GOOO
+            GOOO
+            """).encode()
+        sample_output1 = textwrap.dedent("""\
+            Case #1: 0
+            Case #2: 10
+            """).encode()
+        sample_input2 = textwrap.dedent("""\
+            1
+            6 3
+            RAINBOW
+            FIREBALL
+            RANK
+            RANDOM
+            FIREWALL
+            FIREFIGHTER
+            """).encode()
+        sample_output2 = textwrap.dedent("""\
+            Case #1: 6
+            """).encode()
+        self.assertEqual(problem.download_sample_cases(), [
+            TestCase(
+                'sample-1',
+                'Input 1',
+                sample_input1,
+                'Output 1',
+                sample_output1,
+            ),
+            TestCase(
+                'sample-2',
+                'Input 2',
+                sample_input2,
+                'Output 2',
+                sample_output2,
             ),
         ])
 
@@ -120,10 +171,10 @@ class GoogleCodeJamProblemTest(unittest.TestCase):
             """).encode()
         self.assertEqual(problem.download_sample_cases(), [
             TestCase(
-                'sample',
-                'Input',
+                'sample-1',
+                'Input 1',
                 sample_input,
-                'Output',
+                'Output 1',
                 sample_output,
             ),
         ])

